src/lib/operators/jit_operator/operators/jit_expression.cpp

```diff
@@ -1,13 +1,70 @@
+#define JIT_GET_ENUM_VALUE(index, s) APPEND_ENUM_NAMESPACE(_, _, BOOST_PP_TUPLE_ELEM(3, 1, BOOST_PP_SEQ_ELEM(index, s)))
+
+// Returns the data type (e.g., int32_t, std::string) of a data type defined in the DATA_TYPE_INFO sequence
+#define JIT_GET_DATA_TYPE(index, s) BOOST_PP_TUPLE_ELEM(3, 0, BOOST_PP_SEQ_ELEM(index, s))
+
+// Call the compute function with the correct template parameter if compute is called without a template parameter and
+// store the result in the runtime tuple.
+#define JIT_COMPUTE_CASE(r, types)                                         \
+  case JIT_GET_ENUM_VALUE(0, types): {                                     \
+    const auto result = compute<JIT_GET_DATA_TYPE(0, types)>(context);     \
+    _result_value.set<JIT_GET_DATA_TYPE(0, types)>(result.value, context); \
+    if (_result_value.is_nullable()) {                                     \
+      _result_value.set_is_null(result.is_null, context);                  \
+    }                                                                      \
+    break;                                                                 \
+  }
+
+#define JIT_VARIANT_GET(r, d, type)                          \
+  template <>                                                \
+  BOOST_PP_TUPLE_ELEM(3, 0, type)                            \
+  JitVariant::get<BOOST_PP_TUPLE_ELEM(3, 0, type)>() const { \
+    return BOOST_PP_TUPLE_ELEM(3, 1, type);                  \
+  }
+
+#define JIT_VARIANT_SET(r, d, type)                                                                      \
+  template <>                                                                                            \
+  void JitVariant::set<BOOST_PP_TUPLE_ELEM(3, 0, type)>(const BOOST_PP_TUPLE_ELEM(3, 0, type) & value) { \
+    BOOST_PP_TUPLE_ELEM(3, 1, type) = value;                                                             \
+  }
+
+#define INSTANTIATE_COMPUTE_FUNCTION(r, d, type)                                                              \
+  template JitValue<BOOST_PP_TUPLE_ELEM(3, 0, type)> JitExpression::compute<BOOST_PP_TUPLE_ELEM(3, 0, type)>( \
+      JitRuntimeContext & context) const;
+
+// Instantiate get and set functions for custom JitVariant
+BOOST_PP_SEQ_FOR_EACH(JIT_VARIANT_GET, _, JIT_DATA_TYPE_INFO)
+BOOST_PP_SEQ_FOR_EACH(JIT_VARIANT_SET, _, JIT_DATA_TYPE_INFO)
+
 JitExpression::JitExpression(const JitTupleValue& tuple_value)
     : _expression_type{JitExpressionType::Column}, _result_value{tuple_value} {}
 
+JitExpression::JitExpression(const JitTupleValue& tuple_value, const AllTypeVariant& variant)
+    : _expression_type{JitExpressionType::Value}, _result_value{tuple_value} {
+  _variant.is_null = variant_is_null(variant);
+  if (!_variant.is_null) {
+    resolve_data_type(data_type_from_all_type_variant(variant), [&](const auto current_data_type_t) {
+      using CurrentType = typename decltype(current_data_type_t)::type;
+      _variant.set<CurrentType>(boost::get<CurrentType>(variant));
```

Comment:
If you simply want to get the type of the variant, boost::apply_visitor might be better.